src/Orc.NuGetExplorer/NuGet/Resources/DependencyInfoResourceCollection.cs

```diff
@@ -40,11 +42,26 @@ public DependencyInfoResourceCollection(DependencyInfoResource resource)
             Argument.IsNotNull(() => resource);
 
             _resources = new List<DependencyInfoResource>
-            { 
-                resource 
+            {
+                resource
             };
         }
 
+        public async Task<IEnumerable<SourcePackageDependencyInfo>> ResolvePackagesWithVersionSatisfyRangeAsync(PackageIdentity package, VersionRange versionRange, NuGetFramework projectFramework, SourceCacheContext cacheContext,
+            ILogger log, CancellationToken token)
+        {
+            var singlePackage = await ResolvePackageAsync(package, projectFramework, cacheContext, log, token);
+
+            // Check is this package satisfy requirements, if not, retrieve all dependency infos and find required package
+            if (singlePackage is not null && versionRange.Satisfies(singlePackage.Version))
+            {
+                return new[] { singlePackage };
```

Comment:
maybe add logging so we can trace easily in the future

Reply:
Sure, this is specific case, good to log